contrib/database/sql/sql.go

```diff
@@ -12,49 +12,169 @@
 package sql
 
 import (
+	"context"
 	"database/sql"
 	"database/sql/driver"
 	"errors"
+	"math"
+	"reflect"
+
+	"gopkg.in/DataDog/dd-trace-go.v1/contrib/database/sql/internal"
 )
 
+// registeredDrivers holds a registry of all drivers registered via the sqltrace package.
+var registeredDrivers = &driverRegistry{
+	keys:    make(map[reflect.Type]string),
+	drivers: make(map[string]driver.Driver),
+	configs: make(map[string]*config),
+}
+
+type driverRegistry struct {
+	// keys maps driver types to their registered names.
+	keys map[reflect.Type]string
+	// drivers maps keys to their registered driver.
+	drivers map[string]driver.Driver
+	// configs maps keys to their registered configuration.
+	configs map[string]*config
+}
+
+// registered checks if the named driver has already been registered.
+func (d *driverRegistry) registered(name string) bool {
```

Comment:
Would `isRegistered` be a clearer name and a better indication of what the return value means?

Reply:
Agreed. Changed to `isRegistered`